### DISCUSSION TITLE
Best practice for saving tokens

### DISCUSSION THREAD

**Discussion by stanwrkt** [invertase/react-native-firebase#6068] [Q&A]
Hello, I've followed the official guide about how to set up SDK. It says that it is good practice to get a token during the app launch and then save it to the database.

But this confuses me a little, why don't need to check the user permissions? Why do we need a token if we can't send notifications (= the user didn't grant permissions).

I would like to check what can be a best practice for this case.

**Comment by andersonaddo** [comment]
It's likely it was assumed that you'd ask for permissions first.
No need to get tokens if you don't have the permissions.

Here's an example of how I did it in a previous project:
https://github.com/emitapp/emit-public/blob/276b7b444245622499f4a5e558edb225bdcfe411/src/screens/MainTabNav.js#L173-L207

**Reply by mikehardy** [reply]
yeah, agreed on that - they likely assumed that whatever privacy respecting practices you must follow, were already followed and you obtained permission. More concretely and as an example, if you are respecting the GDPR rules I don't believe it would be permissable to get the token and store it unless the user gave permission to store their data. That is separate from notification permission even
